### PR TITLE
Add support for service accounts in nomad-bridge chart

### PR DESCRIPTION
**Pull request by evfurman**
## SUMMARY
Add support for service accounts in nomad-bridge chart

```yaml
# when Values.serviceAccount.create

❯ helm template .
---
# Source: nomad-bridge/templates/config/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: release-name-nomad-bridge
  labels:
    helm.sh/chart: nomad-bridge-0.1.3
    app.kubernetes.io/name: nomad-bridge
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
  annotations:
    prometheus.io/port: "9090"
    prometheus.io/scrape: "true"
---
# Source: nomad-bridge/templates/processor/processor-statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: release-name-nomad-bridge-processor
  labels:
    helm.sh/chart: nomad-bridge-0.1.3
    app.kubernetes.io/name: nomad-bridge
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: processor
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: nomad-bridge
      app.kubernetes.io/instance: release-name
      app.kubernetes.io/component: processor
  replicas: 1
  serviceName: release-name-nomad-bridge-processor-svc
  template:
    metadata:
      annotations:
        prometheus.io/port: "9090"
        prometheus.io/scrape: "true"
      labels:
        helm.sh/chart: nomad-bridge-0.1.3
        app.kubernetes.io/name: nomad-bridge
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/version: "0.0.1"
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: processor
    spec:
      terminationGracePeriodSeconds: 10
      securityContext:
        fsGroup: 1001
      containers:
        - name: nomad-agent
          image: "gcr.io/nomad-xyz/nomad-agent:latest"
          imagePullPolicy: Always
          securityContext:
            runAsUser: 1001
          command: ["./processor"]
          envFrom:
            - secretRef:
                name: release-name-nomad-bridge-awscreds
          env:
          resources:
            limits:
              cpu: 50m
              memory: 50Mi
            requests:
              cpu: 10m
              memory: 20Mi
          volumeMounts:
            - name: state
              mountPath: /usr/share/nomad
          ports: 
            - name: metrics
              containerPort: 9090
      serviceAccountName: release-name-nomad-bridge
  volumeClaimTemplates:
  - metadata:
      name: state
    spec:
      storageClassName: standard
      accessModes: [ ReadWriteOnce ]
      resources:
        requests:
          storage: 10Gi
---
# Source: nomad-bridge/templates/relayer/relayer-statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: release-name-nomad-bridge-relayer
  labels:
    helm.sh/chart: nomad-bridge-0.1.3
    app.kubernetes.io/name: nomad-bridge
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: relayer
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: nomad-bridge
      app.kubernetes.io/instance: release-name
      app.kubernetes.io/component: relayer
  replicas: 1
  serviceName: release-name-nomad-bridge-relayer-svc
  template:
    metadata:
      annotations:
        prometheus.io/port: "9090"
        prometheus.io/scrape: "true"
      labels:
        helm.sh/chart: nomad-bridge-0.1.3
        app.kubernetes.io/name: nomad-bridge
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/version: "0.0.1"
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: relayer
    spec:
      terminationGracePeriodSeconds: 10
      securityContext:
        fsGroup: 1001
      containers:
        - name: nomad-agent
          image: "gcr.io/nomad-xyz/nomad-agent:latest"
          imagePullPolicy: Always
          securityContext:
            runAsUser: 1001
          command: ["./relayer"]
          envFrom:
            - secretRef:
                name: release-name-nomad-bridge-awscreds
          env:
          resources:
            limits:
              cpu: 50m
              memory: 50Mi
            requests:
              cpu: 10m
              memory: 20Mi
          volumeMounts:
            - name: state
              mountPath: /usr/share/nomad
          ports: 
            - name: metrics
              containerPort: 9090
      serviceAccountName: release-name-nomad-bridge
  volumeClaimTemplates:
  - metadata:
      name: state
    spec:
      storageClassName: standard
      accessModes: [ ReadWriteOnce ]
      resources:
        requests:
          storage: 10Gi
---
# Source: nomad-bridge/templates/updater/updater-statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: release-name-nomad-bridge-updater
  labels:
    helm.sh/chart: nomad-bridge-0.1.3
    app.kubernetes.io/name: nomad-bridge
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: updater
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: nomad-bridge
      app.kubernetes.io/instance: release-name
      app.kubernetes.io/component: updater
  replicas: 1
  serviceName: release-name-nomad-bridge-updater-svc
  template:
    metadata:
      annotations:
        prometheus.io/port: "9090"
        prometheus.io/scrape: "true"
      labels:
        helm.sh/chart: nomad-bridge-0.1.3
        app.kubernetes.io/name: nomad-bridge
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/version: "0.0.1"
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: updater
    spec:
      terminationGracePeriodSeconds: 10
      securityContext:
        fsGroup: 1001
      containers:
        - name: nomad-agent
          image: "gcr.io/nomad-xyz/nomad-agent:latest"
          imagePullPolicy: Always
          securityContext:
            runAsUser: 1001
          command: ["./updater"]
          envFrom:
            - secretRef:
                name: release-name-nomad-bridge-awscreds
          env:
          resources:
            limits:
              cpu: 50m
              memory: 50Mi
            requests:
              cpu: 10m
              memory: 20Mi
          volumeMounts:
            - name: state
              mountPath: /usr/share/nomad
          ports: 
            - name: metrics
              containerPort: 9090
      serviceAccountName: release-name-nomad-bridge
  volumeClaimTemplates:
  - metadata:
      name: state
    spec:
      storageClassName: standard
      accessModes: [ ReadWriteOnce ]
      resources:
        requests:
          storage: 10Gi
```

```yaml
# when ! .Values.serviceAccount.create

❯ helm template .
---
# Source: nomad-bridge/templates/processor/processor-statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: release-name-nomad-bridge-processor
  labels:
    helm.sh/chart: nomad-bridge-0.1.3
    app.kubernetes.io/name: nomad-bridge
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: processor
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: nomad-bridge
      app.kubernetes.io/instance: release-name
      app.kubernetes.io/component: processor
  replicas: 1
  serviceName: release-name-nomad-bridge-processor-svc
  template:
    metadata:
      annotations:
        prometheus.io/port: "9090"
        prometheus.io/scrape: "true"
      labels:
        helm.sh/chart: nomad-bridge-0.1.3
        app.kubernetes.io/name: nomad-bridge
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/version: "0.0.1"
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: processor
    spec:
      terminationGracePeriodSeconds: 10
      securityContext:
        fsGroup: 1001
      containers:
        - name: nomad-agent
          image: "gcr.io/nomad-xyz/nomad-agent:latest"
          imagePullPolicy: Always
          securityContext:
            runAsUser: 1001
          command: ["./processor"]
          envFrom:
            - secretRef:
                name: release-name-nomad-bridge-awscreds
          env:
          resources:
            limits:
              cpu: 50m
              memory: 50Mi
            requests:
              cpu: 10m
              memory: 20Mi
          volumeMounts:
            - name: state
              mountPath: /usr/share/nomad
          ports: 
            - name: metrics
              containerPort: 9090
  volumeClaimTemplates:
  - metadata:
      name: state
    spec:
      storageClassName: standard
      accessModes: [ ReadWriteOnce ]
      resources:
        requests:
          storage: 10Gi
---
# Source: nomad-bridge/templates/relayer/relayer-statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: release-name-nomad-bridge-relayer
  labels:
    helm.sh/chart: nomad-bridge-0.1.3
    app.kubernetes.io/name: nomad-bridge
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: relayer
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: nomad-bridge
      app.kubernetes.io/instance: release-name
      app.kubernetes.io/component: relayer
  replicas: 1
  serviceName: release-name-nomad-bridge-relayer-svc
  template:
    metadata:
      annotations:
        prometheus.io/port: "9090"
        prometheus.io/scrape: "true"
      labels:
        helm.sh/chart: nomad-bridge-0.1.3
        app.kubernetes.io/name: nomad-bridge
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/version: "0.0.1"
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: relayer
    spec:
      terminationGracePeriodSeconds: 10
      securityContext:
        fsGroup: 1001
      containers:
        - name: nomad-agent
          image: "gcr.io/nomad-xyz/nomad-agent:latest"
          imagePullPolicy: Always
          securityContext:
            runAsUser: 1001
          command: ["./relayer"]
          envFrom:
            - secretRef:
                name: release-name-nomad-bridge-awscreds
          env:
          resources:
            limits:
              cpu: 50m
              memory: 50Mi
            requests:
              cpu: 10m
              memory: 20Mi
          volumeMounts:
            - name: state
              mountPath: /usr/share/nomad
          ports: 
            - name: metrics
              containerPort: 9090
  volumeClaimTemplates:
  - metadata:
      name: state
    spec:
      storageClassName: standard
      accessModes: [ ReadWriteOnce ]
      resources:
        requests:
          storage: 10Gi
---
# Source: nomad-bridge/templates/updater/updater-statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: release-name-nomad-bridge-updater
  labels:
    helm.sh/chart: nomad-bridge-0.1.3
    app.kubernetes.io/name: nomad-bridge
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: updater
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: nomad-bridge
      app.kubernetes.io/instance: release-name
      app.kubernetes.io/component: updater
  replicas: 1
  serviceName: release-name-nomad-bridge-updater-svc
  template:
    metadata:
      annotations:
        prometheus.io/port: "9090"
        prometheus.io/scrape: "true"
      labels:
        helm.sh/chart: nomad-bridge-0.1.3
        app.kubernetes.io/name: nomad-bridge
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/version: "0.0.1"
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: updater
    spec:
      terminationGracePeriodSeconds: 10
      securityContext:
        fsGroup: 1001
      containers:
        - name: nomad-agent
          image: "gcr.io/nomad-xyz/nomad-agent:latest"
          imagePullPolicy: Always
          securityContext:
            runAsUser: 1001
          command: ["./updater"]
          envFrom:
            - secretRef:
                name: release-name-nomad-bridge-awscreds
          env:
          resources:
            limits:
              cpu: 50m
              memory: 50Mi
            requests:
              cpu: 10m
              memory: 20Mi
          volumeMounts:
            - name: state
              mountPath: /usr/share/nomad
          ports: 
            - name: metrics
              containerPort: 9090
  volumeClaimTemplates:
  - metadata:
      name: state
    spec:
      storageClassName: standard
      accessModes: [ ReadWriteOnce ]
      resources:
        requests:
          storage: 10Gi
```

### CODE REVIEW

File: charts/nomad-bridge/values.yaml [hunk 1]
```diff
@@ -7,6 +7,14 @@ nameOverride: ""
 #
 fullnameOverride: ""
 
+# serviceAccount -- ServiceAccount configuration
+#
+serviceAccount:
+  # serviceAccount.create -- Create service account
+  create: true
+  # serviceAccount.name -- Name of service account to use if create is false
```
Comment (arnaud036) [inline charts/nomad-bridge/values.yaml:15]:
```suggestion
  # serviceAccount.name -- Name of service account to use if create is true
```

File: charts/nomad-bridge/templates/config/serviceaccount.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "agent.serviceAccountName" . }}
```
Comment (arnaud036) [inline charts/nomad-bridge/templates/config/serviceaccount.yaml:5]: Don't you also need to assign the ServiceAccount to your Pod through the StatefulSets?

**Comment by arnaud036** [comment]
Thanks for your contribution and updating the documentation. :)

File: charts/nomad-bridge/templates/config/serviceaccount.yaml (hunk 2 above)
Comment (evfurman) [reply charts/nomad-bridge/templates/config/serviceaccount.yaml:5]: Added

File: charts/nomad-bridge/templates/processor/processor-statefulset.yaml [hunk 3]
```diff
@@ -66,6 +66,7 @@ spec:
           ports: 
             - name: metrics
               containerPort: {{ .Values.metrics.port }}
+       {{ include "agent.serviceAccountName" . | nindent 6 }}
```
Comment (dflorness-circle) [inline charts/nomad-bridge/templates/processor/processor-statefulset.yaml:69]:
```suggestion
      {{ include "agent.serviceAccountName" . | nindent 6 }}
```

**Review by arnaud036 — APPROVED** [review]

**Pull request merged**